src/WebView.macos.tsx

```diff
@@ -0,0 +1,347 @@
+import React from 'react';
+import {
+  UIManager as NotTypedUIManager,
+  View,
+  requireNativeComponent,
+  NativeModules,
+  Image,
+  findNodeHandle,
+  ImageSourcePropType,
+} from 'react-native';
+import invariant from 'invariant';
+
+import {
+  defaultOriginWhitelist,
+  createOnShouldStartLoadWithRequest,
+  defaultRenderError,
+  defaultRenderLoading,
+} from './WebViewShared';
+import {
+  WebViewErrorEvent,
+  WebViewHttpErrorEvent,
+  WebViewMessageEvent,
+  WebViewNavigationEvent,
+  WebViewProgressEvent,
+  WebViewTerminatedEvent,
+  MacOSWebViewProps,
+  NativeWebViewMacOS,
+  ViewManager,
+  State,
+  RNCWebViewUIManagerMacOS,
+} from './WebViewTypes';
+
+import styles from './WebView.styles';
+
+const UIManager = NotTypedUIManager as RNCWebViewUIManagerMacOS;
+
+const { resolveAssetSource } = Image;
+
+const RNCWebViewManager = NativeModules.RNCWebViewManager as ViewManager;
+
+const RNCWebView: typeof NativeWebViewMacOS = requireNativeComponent(
+  'RNCWebView',
+);
+
+class WebView extends React.Component<MacOSWebViewProps, State> {
+  static defaultProps = {
+    javaScriptEnabled: true,
+    cacheEnabled: true,
+    originWhitelist: defaultOriginWhitelist,
+    useSharedProcessPool: true,
+  };
+
+  static isFileUploadSupported = async () => {
+    // no native implementation for macOS, depends only on permissions
+    return true;
```

Comment:
Should this be false?

Reply:
I assumed it would be the same as iOS.   On Android there is an actual native module method that gets called.   On iOS (and now macOS) there is no native method, so it just returns true.

Reply:
Sounds good.